### PR TITLE
Fix imports github.com/aws/amazon-cloudwatch-agent-test/internal/comm…

### PR DESCRIPTION
**Pull request by sethAmazon**
…on: build constraints exclude all Go files in /home/ec2-user/amazon-cloudwatch-agent-test/internal/common

# Description of the issue
Weird edge case where if there is a //go:build unix not allowing a call from linux env when the original file sanity_test.go does not contain a unix build tag but call a *_unix.go file. 
```
null_resource.integration_test (remote-exec): 	imports github.com/aws/amazon-cloudwatch-agent-test/internal/common: build constraints exclude all Go files in /home/ec2-user/amazon-cloudwatch-agent-test/internal/common
null_resource.integration_test (remote-exec): no Go files in /home/ec2-user/amazon-cloudwatch-agent-test
╷
│ Error: remote-exec provisioner error
│ 
│   with null_resource.integration_test,
│   on main.tf line 120, in resource "null_resource" "integration_test":
│  120:   provisioner "remote-exec" {
│ 
│ error executing "/tmp/terraform_684748912.sh": Process exited with status 1
```

# Description of changes
Remove internal package so it can be imported in other packages. 

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
```
[ec2-user@ip-172-31-45-254 amazon-cloudwatch-agent-test]$ go test ./test/sanity -p 1 -v
=== RUN   TestAgentStatus
--- PASS: TestAgentStatus (2.37s)
PASS
ok  	github.com/aws/amazon-cloudwatch-agent-test/test/sanity	2.370s
```

```
[ec2-user@ip-172-31-45-254 amazon-cloudwatch-agent-test]$ go test ./test/collection_interval -p 1 -timeout 1h -computeType=EC2 -v
...
--- PASS: TestCollectionInterval (717.14s)
    --- PASS: TestCollectionInterval/test_description_No_collection_interval_given_default_to_60s_resource_file_location_resources/default.json_number_of_metrics_lower_bound_1_and_upper_bound_3 (177.14s)
    --- PASS: TestCollectionInterval/test_description_Agent_has_10s_second_collection_interval_resource_file_location_resources/agent_interval_10s.json_number_of_metrics_lower_bound_11_and_upper_bound_13 (179.90s)
    --- PASS: TestCollectionInterval/test_description_Metric_disk_has_10s_collection_interval_resource_file_location_resources/metric_interval_10s.json_number_of_metrics_lower_bound_11_and_upper_bound_13 (180.03s)
    --- PASS: TestCollectionInterval/test_description_Agent_has_60s_collection_interval,_disk_has_10s_collection_interval_use_disk_collection_interval_resource_file_location_resources/metric_override_interval_10s.json_number_of_metrics_lower_bound_11_and_upper_bound_13 (180.06s)
PASS
ok  	github.com/aws/amazon-cloudwatch-agent-test/test/collection_interval	717.142s
```

### CODE REVIEW

File: msi/tools/msiversion/msiversionconverter.go [hunk 1]
```diff
@@ -2,8 +2,6 @@
 // SPDX-License-Identifier: MIT
 
 //go:build unix
-// +build unix
```
Comment (sethAmazon) [inline msi/tools/msiversion/msiversionconverter.go:5]: // +build unix is no longer advised by golang. It can still functional code.

File: internal/common/agent_util_unix.go [hunk 2]
```diff
@@ -1,8 +1,7 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: MIT
 
-//go:build unix
-// +build unix
+//go:build unix || linux
```
Comment (sethAmazon) [inline internal/common/agent_util_unix.go:4]: Weird edge case where if there is a //go:build unix not allowing a call from linux env when the original file sanity_test.go does not contain a unix build tag but call a *_unix.go file.

File: test/metric_value_benchmark/disk_test.go [hunk 3]
```diff
@@ -15,34 +16,36 @@ import (
 )
 
 type DiskTestRunner struct {
```
Comment (sethAmazon) [inline test/metric_value_benchmark/disk_test.go:18]: This test can now compile

File: test/sanity/sanity_unix.go [hunk 4]
```diff
@@ -1,6 +1,8 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: MIT
 
+//go:build unix || linux
```
Comment (adam-mateen) [inline test/sanity/sanity_unix.go:4]: Isn't linux a subset of unix?
Why not just:
```
//go:build unix
```
?

Comment (sethAmazon) [reply test/sanity/sanity_unix.go:4]: Weird edge case where if there is a //go:build unix not allowing a call from linux env when the original file sanity_test.go does not contain a unix build tag but call a *_unix.go file.

For some reason, it gives error 
```
null_resource.integration_test (remote-exec): 	imports github.com/aws/amazon-cloudwatch-agent-test/internal/common: build constraints exclude all Go files in /home/ec2-user/amazon-cloudwatch-agent-test/internal/common
null_resource.integration_test (remote-exec): no Go files in /home/ec2-user/amazon-cloudwatch-agent-test
╷
│ Error: remote-exec provisioner error
│ 
│   with null_resource.integration_test,
│   on main.tf line 120, in resource "null_resource" "integration_test":
│  120:   provisioner "remote-exec" {
│ 
│ error executing "/tmp/terraform_684748912.sh": Process exited with status 1
```

This change resolved the issues. I think there is something broken for this edge case in the golang compiler.

File: internal/common/agent_util_unix.go (hunk 2 above)
Comment (SaxyPandaBear) [reply internal/common/agent_util_unix.go:4]: We should just change it to !windows.

File: generator/test_case_generator.go [hunk 5]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 type matrixRow struct {
-	TestDir             string `json:"testDir"`
+	TestDir             string `json:"test_dir"`
```
Comment (sethAmazon) [inline generator/test_case_generator.go:16]: I messed up the dir name. This fixes that.

File: test/metric/dimension/custom_provider.go [hunk 6]
```diff
@@ -28,6 +27,6 @@ func (p *CustomDimensionProvider) GetDimension(instruction Instruction) types.Di
 
 	return types.Dimension{
 		Name:  aws.String(instruction.Key),
-		Value: aws.String(*instruction.Value.Value),
```
[the remark this answers is not in the document]
Comment (SaxyPandaBear) [reply test/metric/dimension/custom_provider.go:31]: The issue is the if condition in `isKnown` is inverted.

File: generator/test_case_generator.go (hunk 5 above)
Comment (khanhntd) [reply generator/test_case_generator.go:16]: **Nitpick**: Other are not consistent with the format (e.g testType - test_type)

File: test/metric_value_benchmark/disk_test.go [hunk 7]
```diff
@@ -15,34 +16,36 @@ import (
...
 	return "Disk"
 }
 
-func (t *DiskTestRunner) getAgentConfigFileName() string {
+func (t *DiskTestRunner) GetAgentConfigFileName() string {
 	return "disk_config.json"
 }
-func (t *DiskTestRunner) getAgentRunDuration() time.Duration {
-	return minimumAgentRuntime
+func (t *DiskTestRunner) GetAgentRunDuration() time.Duration {
```
Comment (khanhntd) [inline test/metric_value_benchmark/disk_test.go:44]: Every other sanity test shares the same `GetAgentRunDuration` and also the `MinimumAgentRuntime`. Therefore, would it be better to add GetAgentRunDuration() in `BaseTestRunner`?

**Review by khanhntd — APPROVED** [review]

File: test/metric_value_benchmark/disk_test.go (hunk 7 above)
Comment (williazz) [reply test/metric_value_benchmark/disk_test.go:44]: Yeah I would second this nit pick

**Review by williazz — APPROVED** [review]
